scaaml/capture/aes_capture.py

```diff
@@ -0,0 +1,206 @@
+from pathlib import Path
+
+from scaaml.capture.aes_capture_util import *
+from scaaml.capture.acqktp import AcqKeyTextPattern_Scaaml as ktp_scaaml
+from scaaml.io import Dataset
+from scaaml.io.resume_kti import ResumeKTI
+
+
+class AESCaptureRunner:
+    def __init__(
+        self,
+        firmware_hash:
+        str = '18ba5773bbd766d5e6f76a6f1ebb689570ebed679232e22c8cf9ddbd140196d1',
+        architecture: str = "CW308_STM32F4",
+        implementation: str = "MBEDTLS",
+        algorithm: str = "simpleserial-aes",
+        version: int = 1,
+        root_path: str = "/mnt/storage/chipwhisperer",
+        dataset: str = "Training",
+        purpose: str = "train",
+        keys: int = 3072,
+        plaintexts: int = 256,
+        repetitions: int = 1,
+        samples: int = 7000,
+        offset: int = 0,
+        clock: int = 7372800,
+        gain: int = 45,
+        sample_rate: str = "clkgen_x4",
+        capture_info={},
```

Comment:
Don't put mutable default values: they create a hidden class variable which can lead to unwanted side effects between object instances

Reply:
I will try to remember this.